src/robot_module/src/locks/lock_service_client.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+
+from robot_module.srv import acquire_lock, release_lock
+from std_msgs.msg import String
+import rospy
+import time
+
+class LockClient:
+    def __init__(self, name):
+        self.name = name
+
+
+    def __enter__(self):
+        acquire_attempts = 0
+        lock_acquired = self.lock()
+
+        # use exponential backoff to try again without flooding network
+        while not lock_acquired:
+            rospy.loginfo("Couldn't acquire the lock, trying again")

```

Comment:
Consider changing to a debug message instead, this will pop up whenever a node is waiting on the lock every once in a while.